src/events.rs

```diff
@@ -338,6 +346,229 @@ impl EventMask {
     }
 }
 
+/// A struct that provides structured access to event masks
+/// returned from reading an event from an inotify fd
+#[derive(Debug)]
+pub struct ParsedEventMask {
+    /// The kind of event that occurred
+    pub kind: Option<EventKind>,
+    /// The auxillary flags about the event
+    pub auxillary_flags: EventAuxillaryFlags,
+}
+
+impl ParsedEventMask {
+    /// Construct a `ParsedEventMask` from its component parts
+    pub fn from_parts(kind: Option<EventKind>, auxillary_flags: EventAuxillaryFlags) -> Self {
+        ParsedEventMask {
+            kind,
+            auxillary_flags,
+        }
+    }
+
+    /// Parse a raw event mask
+    pub fn from_raw_event_mask(mask: EventMask) -> Result<Self, EventMaskParseError> {
+        if mask.contains(EventMask::Q_OVERFLOW) {
+            return Err(EventMaskParseError::QueueOverflow);
+        }
+
+        let kind = mask.try_into()?;
+        let auxillary_flags = mask.into();
+
+        Ok(ParsedEventMask::from_parts(kind, auxillary_flags))
+    }
+}
+
+impl TryFrom<EventMask> for ParsedEventMask {
+    type Error = EventMaskParseError;
+
+    fn try_from(value: EventMask) -> Result<Self, Self::Error> {
+        Self::from_raw_event_mask(value)
+    }
+}
+
+/// Represents the type of inotify event
+///
+/// Exactly 0 or 1 of these bitflags will be set in an event mask
+/// returned from reading an inotify fd
```

Comment:
I think this is slightly confusing for someone not knowing the full context, as it refers to bitflags, when this is an enum. Might be better to remove it here and instead add this information to the `kind: Option<EventKind>` field instead. Specifically the `0 or 1` part makes more sense there, with the `Option` directly beside it.